src/integration/search-index.ts

```diff
@@ -0,0 +1,84 @@
+import type { AstroIntegration} from "astro";
+import { readFileSync, writeFileSync } from "fs";
+import { JSDOM } from "jsdom";
+
+
+function* extractText(node: Node): Generator<string, void, void> {
+    if (node.hasChildNodes()) {
+        for (const c of node.childNodes) {
+            for (const r of extractText(c)) {
+                yield r;
+            }
+        }
+    } else {
+        yield node.textContent!;
+    }
+}
+
+function cleanPath(path: string): string {
+    if (process.platform === "win32") {
```

Comment:
why not simply check if it starts with `/`?

Reply:
Linux

Reply:
but your comment says only on windows do the paths start with an extra `/`?

Reply:
On windows it looks like `/C:/...` which is just.. wrong. But on Linux you have your normal `/home/...` path, if you remove the leading slash here you get a broken path as well.

Reply:
Why not relativize the paths before returning them?
is that not what that `dir` parameter is for?

Reply:
No, because the path is fucked and needs to be unfucked no matter what. This works, honestly.